app/gui2/src/util/qualifiedName.ts

```diff
@@ -64,7 +70,7 @@ export function qnJoin(left: QualifiedName, right: QualifiedName): QualifiedName
  * The element is considered a top element if there is max 1 segment in the path.
  */
 export function qnIsTopElement(name: QualifiedName): boolean {
-  return (name.match(/\./g)?.length ?? 0) <= 2
+  return !/[.].*?[.].*?[.]/.test(name)
```

Comment:
I'm ok with this change, only I'm curious why it is here? It is more performant, or something?

Reply:
i haven't confirmed whether it's more important, but: it doesn't have to store the match strings, nor create an array, nor are there any conditionals caused by optional chaining
so at the very least i don't think it's *less* performant